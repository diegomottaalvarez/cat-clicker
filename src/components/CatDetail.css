.cat-detail {
  display: flex;
}

.cat-img-container,
.cat-description-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 1rem;
}

.cat-img-container {
  position: relative;
}

button.leave-cat-button {
  width: 30rem;
  font-size: 2rem;
  transition: all 0.3s ease-in-out;
  background-color: #006983;
  color: #ffffff;
}

.breed-name {
  font-size: 2rem;
}

@media (min-width: 901px) {
  button.leave-cat-button:hover {
    transform: scale(1.1);
    background-color: #5ab7d4;
    color: rgb(41, 41, 41);
  }

  .breed-name {
    margin-top: 0;
  }
}

@media (max-width: 900px) {
  .cat-detail {
    flex-direction: column;
  }

  button.leave-cat-button {
    width: 100%;
  }
}
