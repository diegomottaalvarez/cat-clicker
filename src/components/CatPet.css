.pet-button {
  width: 30rem;
  font-size: 2rem;
  display: flex;
  justify-content: space-evenly;
  transition: all 0.3s ease-in-out;
}

@media (min-width: 900px) {
  .pet-button:hover {
    transform: scale(1.1);
    color: #ffffff;
  }
}

.pet-button img {
  width: 2rem;
}

.heart-counter-container {
  display: flex;
  align-items: flex-start;
}

@media (max-width: 900px) {
  .pet-button {
    width: 100%;
  }
}
