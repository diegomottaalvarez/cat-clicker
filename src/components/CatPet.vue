<template>
  <button class="pet-button" data-testid="increaseBtn" @click="addClick"
    >Pet me!
    <div class="heart-counter-container"
      ><img src="@/assets/icons/heart.svg" alt="heart-icon" />
      <span data-testid="clickCounter">{{ catClicks }}</span></div
    >
  </button>
</template>
<script>
export default {
  props: {
    catClicks: 0,
  },
  methods: {
    addClick() {
      this.$emit('catPetClick');
    },
  },
};
</script>
<style scoped>
@import './CatPet.css';
</style>
