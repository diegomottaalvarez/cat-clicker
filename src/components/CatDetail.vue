<template>
  <div v-if="catDetail" class="cat-detail">
    <div class="cat-img-container">
      <button class="leave-cat-button" @click="$emit('unselectCat')"
        >Leave me alone</button
      >
      <CatImage :catDetail="catDetail" @catPetClick="addClick" />
      <CatPet :catClicks="catDetail.catClicks || 0" @catPetClick="addClick" />
    </div>
    <div class="cat-description-container">
      <h2 class="breed-name">{{ catDetail.breeds[0].name }}</h2>
      <p class="cat-info">{{ catDetail.breeds[0].description }}</p>
      <a
        v-if="catDetail.breeds[0].vcahospitals_url"
        class="link"
        :href="catDetail.breeds[0].vcahospitals_url"
        target="_blank"
        >Tell me more!</a
      >
    </div>
  </div>
</template>
<script>
import CatPet from './CatPet.vue';
import CatImage from './CatImage.vue';
export default {
  props: {
    catDetail: { type: Object },
  },
  methods: {
    addClick() {
      this.catDetail.catClicks
        ? this.catDetail.catClicks++
        : (this.catDetail.catClicks = 1);
    },
  },
  components: { CatPet, CatImage },
};
</script>
<style scoped>
@import './CatDetail.css';
</style>
