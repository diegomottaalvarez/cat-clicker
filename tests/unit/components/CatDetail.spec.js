import { shallowMount } from '@vue/test-utils';
import CatDetail from '@/components/CatDetail';
import { catDetailMock } from '@/helpers/getCats';
import CatPet from '@/components/CatPet';
describe('CatDetail component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(CatDetail, {
      propsData: {
        catDetail: catDetailMock,
      },
    });
  });
  test('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should add 1 when click', async () => {
    const child = wrapper.findComponent(CatPet);
    let catClicks = Number(wrapper.vm.catDetail.catClicks);
    if (!catClicks) catClicks = 0;
    await child.vm.$emit('catPetClick');
    const catClicksAfter = Number(wrapper.vm.catDetail.catClicks);
    expect(catClicksAfter).toBe(catClicks + 1);
  });
});
