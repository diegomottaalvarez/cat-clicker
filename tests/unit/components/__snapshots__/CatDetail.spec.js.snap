// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatDetail component should match the snapshot 1`] = `
<div class="cat-detail">
  <div class="cat-img-container"><button class="leave-cat-button">Leave me alone</button>
    <cat-image-stub catdetail="[object Object]"></cat-image-stub>
    <cat-pet-stub catclicks="0"></cat-pet-stub>
  </div>
  <div class="cat-description-container">
    <h2 class="breed-name">Somali</h2>
    <p class="cat-info">The Somali lives life to the fullest. He climbs higher, jumps farther, plays harder. Nothing escapes the notice of this highly intelligent and inquisitive cat. Somalis love the company of humans and other animals.</p><a class="link" href="https://vcahospitals.com/know-your-pet/cat-breeds/somali" target="_blank">Tell me more!</a>
  </div>
</div>
`;
